src/typed-method-types/workflows/triggers/mod.ts

```diff
@@ -30,44 +58,90 @@ export type BaseTrigger = {
   description?: string;
 };
 
+// A helper to make sure inputs are passed. Required for automated triggers
+export type RequiredInputs = Required<
+  Pick<BaseTrigger, "inputs">
+>;
+
+/** @description Response object content for delete method */
+type DeleteResponse = {
+  ok: true;
+};
+
+type DeleteTriggerResponse = Promise<
+  DeleteResponse | FailedTriggerResponse
+>;
+
+type ListArgs = {
+  /** @description Lists triggers only if they owned by the caller */
+  is_owner?: boolean;
+  /** @description Lists triggers only if they have been published */
+  is_published?: boolean;
+};
+
+type ValidTriggerObjects =
+  | ShortcutTriggerObject<ShortcutTrigger>
+  | EventTriggerObject<EventTrigger>
+  | ScheduledTriggerObject<ScheduledTrigger>
+  | WebhookTriggerObject<WebhookTrigger>;
+
+type ListResponse = {
+  ok: true;
+  triggers: ValidTriggerObjects[];
+};
+
+type ListTriggerResponse = Promise<
+  ListResponse | FailedListResponse
+>;
+
+export type FailedListResponse = BaseResponse & {
+  ok: false;
+  /** @description no triggers are returned on a failed response */
+  triggers?: never;
+};
+
+export type FailedTriggerResponse = BaseResponse & {
+  ok: false;
+  /** @description no trigger is returned on a failed response */
+  trigger?: never;
+};
+
 export type TriggerIdType = {
   /** @description The id of a specified trigger */
-  trigger_id: string;
+  trigger_id: string | undefined;
```

Comment:
It looks like `TriggerIdType` is used for our `update` and `delete` methods. It feels like the `trigger_id` would be required when using either of those?